### PR TITLE
ICU-21552 Only show the "Encountered empty table warning" in verbose mode

### PR DESCRIPTION
**Pull request by richgillam**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21552
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [ ] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

File: icu4c/source/tools/genrb/parse.cpp [hunk 1]
```diff
@@ -1191,9 +1194,11 @@ realParseTable(ParseState* state, TableResource *table, char *tag, uint32_t star
 
         if (token == TOK_CLOSE_BRACE)
         {
+#if WARN_EMPTY_TABLE
             if (!readToken) {
                 warning(startline, "Encountered empty table");
             }
+#endif
```
Comment (markusicu) [inline icu4c/source/tools/genrb/parse.cpp:1201]:
```suggestion
            if (!readToken && isVerbose()) {
                warning(startline, "Encountered empty table");
            }
```
... is a better fix I think.

Comment (richgillam) [reply icu4c/source/tools/genrb/parse.cpp:1201]: \*sigh\* You're right.  And my question to get one of these simple changes through in one try continues...

**Review by markusicu — COMMENTED** [review]
change lgtm

please squash, and fix the commit comment

**Review by markusicu — APPROVED** [review]
lgtm tnx

**Comment by markusicu** [comment]
... and... I did it again :-(
> Merged #1663 into master.

@sffc @jefgen FYI

I had intended for this to go into ICU 69, which means it should have gone into maint/maint-69, but once more I spaced out and it went into master. Now we have to decide whether to cherry-pick it into 69 (probably not worth it) or move the ticket to 70.

Sorry... Unrealistic wish list: Once we have a maintenance branch, I wish there was something that asked "do you really want to merge into master/main?"

**Comment by richgillam** [comment]
Dammit.  I forgot all about that.  I think I'd propose we just move the ticket to ICU 70 and call it a day.

**Comment by richgillam** [comment]
But of course, if enough of you want this change in ICU 69, I'll try to figure out how to cherry-pick it back to the main branch.  I still have PTSD from the last time I tried to do that, but...

**Comment by richgillam** [comment]
>Sorry... Unrealistic wish list: Once we have a maintenance branch, I wish there was something that asked "do you really want to merge into master/main?"

That would definitely be awesome!

**Comment by markusicu** [comment]
I changed the ticket to 70. Not worth the cherry-pick trouble, I think. If someone disagrees, change it back.

Congrats to Rich for (probably?) the first change into ICU 70! :-}

**Comment by sffc** [comment]
This will get handled as part of BRS line 31 ["Check for Missing Commits"](https://sites.google.com/site/icusite/repository/gitdev#TOC-Checking-for-Missing-Commits-BRS-Task-).  I will do a bulk cherry-pick of things that got merged to the wrong place before we tag ICU 69 GA.

**Comment by richgillam** [comment]
Apologies to all for screwing this up; I agree with Markus that it’d be nice if there were some kind of extra confirmation step that would happen automatically when there are two branches active at the same time to make sure you’re checking into the right place.  But mostly it’s just my forgetting because I haven’t been checking thing into ICU often enough to get the process down.

Shane, if this can be one of the things you bulk cherry-pick back to 69, I think that would be lovely.  Would that mean we should move the ticket back to 69 to ensure that happens?

—Rich

**Comment by sffc** [comment]
No reason at all to feel disappointed/ashamed! This is why we have BRS line 31.

In terms of tooling to remind you when you need to merge to maint, I'm not sure how we'd do this in the GitHub framework, but we could probably add it to the bot. https://github.com/unicode-org/jira-github-pr-check/issues/32

I'll fix the fixVersion later.

**Pull request merged**